web/packages/core/src/internal/player/inner.tsx

```diff
@@ -855,6 +856,16 @@ export class InnerPlayer {
                     this.loadedConfig.backgroundColor;
             }
 
+            // We may theoretically need to check everything listed on https://github.com/rust-lang/rust/blob/master/src/doc/rustc/src/platform-support/wasm32-unknown-unknown.md#enabled-webassembly-features
+            // but this is the only extension I know completely breaks our WASM module if unsupported
+            const necessaryExtensionsSupported: boolean = await referenceTypes();
+            if (!necessaryExtensionsSupported) {
```

Comment:
https://github.com/danielhjacobs/ruffle/compare/throw-direct-error-when-reference-types-unsupported...danielhjacobs:ruffle:make-embed-if-flash-supported-ruffle-not

This would create Flash embeds if reference-types is unsupported but Flash is supported. I mainly just had fun making it, as it was interesting to try to let DataLoadOptions work too.